docs/typescript/examples/checkout-tutorial.md

```diff
@@ -1,299 +1,370 @@
+    // Otherwise, return reserved inventory and cancel the order.
+    if (notification && notification === 'paid') {
+      DBOS.logger.info(`Payment successful!`);
+      await ShopUtilities.markOrderPaid(orderID);
+      await DBOS.startWorkflow(ShopUtilities).dispatchOrder(orderID);
+    } else {
+      DBOS.logger.warn(`Payment failed...`);
+      await ShopUtilities.errorOrder(orderID);
+      await ShopUtilities.undoSubtractInventory();
+    }
+
+    // Finally, send the order ID to the payment endpoint so it can redirect
+    // the customer to the order status page.
+    await DBOS.setEvent(ORDER_ID_EVENT, orderID);
+  }
+}
+```
 
-## The Request Handler
+## The Checkout and Payment Endpoints
 
-We'll start by building the checkout request handler, which initiates checkout in response to customer HTTP requests.
+Now, let's use Fastify to write the HTTP endpoint for checkout.
 
-### Registering the handler
+This endpoint receives a request when a customer presses the "Buy Now" button.
+It starts the checkout workflow in the background, then waits for the workflow to generate and send it a unique payment ID.
+It then returns the payment ID so the browser can redirect the user to the payments page.
 
-The handler is implemented in this `webCheckout` function and served from HTTP POST requests to the URL `<host>/checkout/:key?`.
+The endpoint accepts an [idempotency key](../tutorials/workflow-tutorial.md#workflow-ids-and-idempotency) so that even if the customer presses "buy now" multiple times, only one checkout workflow is started.
 
 ```javascript
-@DBOS.postApi('/checkout/:key?')
-static async webCheckout(@ArgOptional key: string): Promise<string> {
-```
+const fastify = Fastify({logger: true});
+
+fastify.post<{
+  Params: { key: string };
+}>('/checkout/:key', async (req, reply) => {
+  const key = req.params.key;
+  // Idempotently start the checkout workflow in the background.
+  const handle = await DBOS.startWorkflow(Shop, { workflowID: key }).paymentWorkflow();
+  // Wait for the checkout workflow to send a payment ID, then return it.
+  const paymentID = await DBOS.getEvent<string | null>(handle.workflowID, PAYMENT_ID_EVENT);
+  if (paymentID === null) {
+    DBOS.logger.error('checkout failed');
+    return reply.code(500).send('Error starting checkout');
+  }
+  return paymentID;
+});
 
-It accepts an optional parameter `key`, used to invoke the checkout workflow [idempotently](../tutorials/workflow-tutorial.md#workflow-ids-and-idempotency).
-If a workflow is invoked many times with the same idempotency key (for example, because a customer pressed the buy button many times), it only executes once.
+```
 
-### Invoking the checkout workflow
-Upon receiving a request, the handler asynchronously invokes the checkout workflow using its idempotency key.
-It obtains a [workflow handle](../reference/transactapi/workflow-handles), used to interact with the workflow.
+Let's also write the HTTP endpoint for payments.
```

Comment:
This is the payment webhook not the actual payment API. We can say this webhook listens on any payment event generated by the user, such as paid or canceled, and then signals the checkout workflow.